run_api.py

```diff
@@ -0,0 +1,105 @@
+import sys
+import os
+import shutil
+from fastapi import FastAPI, File, UploadFile, Depends, Response
+from fastapi.responses import FileResponse
+import uvicorn
+from pydantic import BaseModel
+from typing import Optional, Literal
+
+import roop.globals
+from roop.core import decode_execution_providers, suggest_execution_threads, limit_resources, start, \
+    get_frame_processors_modules, suggest_execution_providers
+from roop.utilities import resolve_relative_path
+
+
+app = FastAPI()
+
+class RoopModel(BaseModel):
+    frame_processor: Optional[list] = ['face_swapper']
+    keep_fps: Optional[bool] = False
+    keep_frames: Optional[bool] = False
+    skip_audio: Optional[bool] = False
+    many_faces: Optional[bool] = False
+    reference_face_position: Optional[int] = 0
+    reference_frame_number: Optional[int] = 0
+    similar_face_distance: Optional[float] = 0.85
+    temp_frame_format: Optional[Literal['png','jpg']] = 'png'
+    temp_frame_quality: Optional[int] = 0
+    output_video_encoder: Optional[Literal['libx264', 'libx265', 'libvpx-vp9', 'h264_nvenc', 'hevc_nvenc']] = 'libx264'
+    output_video_quality: Optional[int] = 0
+    max_memory: Optional[int] = 0
+    execution_threads: Optional[int] = suggest_execution_threads()
+@app.get('/get_execution_providers')
+async def get_execution_provider():
+    return suggest_execution_providers()
+
+
+@app.post("/start_roop")
+async def image_file(
+        src_file: UploadFile = File(...),
+        target_file: UploadFile = File(...),
+        roop_parameters: RoopModel = Depends()
+):
+    # Removing the folder and its content if it already exists
+    saving_path = resolve_relative_path('../workdir/')
+    if os.path.exists(saving_path):
+        shutil.rmtree(saving_path)
+    os.makedirs(saving_path)
+
+    #Get execution provider from env.
```

Comment:
I don't understand why this endpoint exists to be honest... from that logic a frame processors endpoint would be needed aswell 😁

Either you can explain why this ENV do exists or you remove them all together. I would prefer to have an API without limitations.